common/aws/s3/s3.go

```diff
@@ -2,10 +2,47 @@ package s3
 
 import "context"
 
+// Client encapsulates the functionality of an S3 client.
 type Client interface {
-	CreateBucket(ctx context.Context, bucket string) error
+
+	// DownloadObject downloads an object from S3.
 	DownloadObject(ctx context.Context, bucket string, key string) ([]byte, error)
+
+	// UploadObject uploads an object to S3.
```

Comment:
Thank you for adding these comments everywhere!

Reply:
Documenting every function is one of my pet peeves. :)